Sources/AsyncAlgorithms/Channels/ChannelStorage.swift

```diff
@@ -0,0 +1,149 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Async Algorithms open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+final class ChannelStorage<Element, Failure: Error>: @unchecked Sendable {
+  private let stateMachine: ManagedCriticalState<ChannelStateMachine<Element, Failure>>
+  private let ids = ManagedCriticalState<Int>(0)
+
+  init() {
+    self.stateMachine = ManagedCriticalState(ChannelStateMachine())
+  }
+
+  func generateId() -> Int {
+    self.ids.withCriticalRegion { ids in
+      defer { ids += 1 }
```

Comment:
this should probably be `&+=`

```suggestion
      defer { ids &+= 1 }
```

Reply:
done